pkg/rbd/nodeserver.go

```diff
@@ -303,9 +303,25 @@ func (ns *NodeServer) mountVolumeToStagePath(ctx context.Context, req *csi.NodeS
 	// Publish Path
 	fsType := req.GetVolumeCapability().GetMount().GetFsType()
 	diskMounter := &mount.SafeFormatAndMount{Interface: ns.mounter, Exec: mount.NewOsExec()}
+	existingFormat, err := diskMounter.GetDiskFormat(devicePath)
+	if err != nil {
+		return err
+	}
+	if existingFormat == "" && (fsType == "ext4" || fsType == "xfs") {
+		args := []string{}
+		if fsType == "ext4" {
+			args = []string{"-m0", "-Enodiscard", devicePath}
+		} else if fsType == "xfs" {
+			args = []string{"-K", devicePath}
+		}
+		_, err = diskMounter.Exec.Run("mkfs."+fsType, args...)
+		if err != nil {
+			return err
```

Comment:
please log the error message before returning

Reply:
Done.